lib/byron/test/integration/Main.hs

```diff
@@ -0,0 +1,157 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Main where
+
+import Prelude
+
+import Cardano.BM.Data.Severity
+    ( Severity (..) )
+import Cardano.BM.Trace
+    ( Trace, logInfo )
+import Cardano.CLI
+    ( Port (..), withLogging )
+import Cardano.Launcher
+    ( ProcessHasExited (..) )
+import Cardano.Startup
+    ( withUtf8Encoding )
+import Cardano.Wallet.Api.Server
+    ( Listen (..) )
+import Cardano.Wallet.Api.Types
+    ( ApiByronWallet, WalletStyle (..) )
+import Cardano.Wallet.Byron
+    ( serveWallet, setupTracers, tracerSeverities )
+import Cardano.Wallet.Byron.Compatibility
+    ( Byron )
+import Cardano.Wallet.Byron.Config
+    ( withCardanoNode )
+import Cardano.Wallet.Byron.Faucet
+    ( initFaucet )
+import Cardano.Wallet.Network.Ports
+    ( unsafePortNumber )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( NetworkDiscriminant (..) )
+import Cardano.Wallet.Primitive.Types
+    ( SyncTolerance (..) )
+import Control.Concurrent.Async
+    ( race )
+import Control.Concurrent.MVar
+    ( newEmptyMVar, putMVar, takeMVar )
+import Control.Exception
+    ( throwIO )
+import Control.Monad
+    ( forM_, void )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Text
+    ( Text )
+import Network.HTTP.Client
+    ( defaultManagerSettings
+    , managerResponseTimeout
+    , newManager
+    , responseTimeoutMicro
+    )
+import Numeric.Natural
+    ( Natural )
+import System.IO.Temp
+    ( withSystemTempDirectory )
+import Test.Hspec
+    ( Spec, SpecWith, after, describe, hspec )
+import Test.Hspec.Extra
+    ( aroundAll )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , KnownCommand (..)
+    , Payload (..)
+    , TxDescription (..)
+    , request
+    , unsafeRequest
+    )
+
+import qualified Cardano.Wallet.Api.Link as Link
+import qualified Data.Aeson as Aeson
+import qualified Data.Text as T
+import qualified Test.Integration.Scenario.API.ByronTransactions as ByronTransactions
+import qualified Test.Integration.Scenario.API.ByronWallets as ByronWallets
+import qualified Test.Integration.Scenario.API.Network as Network
+
+-- | Define the actual executable name for the bridge CLI
+instance KnownCommand Byron where
+    commandName = "cardano-wallet-byron"
+
+main :: forall t. (t ~ Byron) => IO ()
+main = withUtf8Encoding $ withLogging Nothing Info $ \(_, tr) -> do
+    hspec $ do
+        describe "API Specifications" $ specWithServer tr $ do
+            ByronWallets.spec
+            ByronTransactions.spec
+            Network.spec
+
+specWithServer
+    :: Trace IO Text
+    -> SpecWith (Context Byron)
+    -> Spec
+specWithServer tr = aroundAll withContext . after tearDown
+  where
+    withContext :: (Context Byron -> IO ()) -> IO ()
+    withContext action = do
```

Comment:
would be good to reuse somehow : https://github.com/input-output-hk/cardano-wallet/blob/master/lib/jormungandr/test/integration/Main.hs#L164 if possible and maybe put it somewhere in core-integration

Reply:
Yeah, though the annoying part is that, despite the functions having the same names, they have subtle differences and I am quite reluctant in trying to unify them into one common abstraction :/ 
I am afraid of the complexity this would introduce. Maybe worth the try however.